### DISCUSSION TITLE
duplicate connections on iOS

### DISCUSSION THREAD

**Discussion by ajkahl** [PhilipsHue/flutter_reactive_ble#425] [Q&A]
Hi,

I've a device that allows a couple of concurrent connections - which are usually a combination of a phone and a keyfob remote, or maybe a couple of phones. I've recently moved my app from flutter blue and have it mostly working well, bar one issue. 

I first scan for the device, and if I come across one that has been saved (using an ID in the ad packet to avoid the iOS UUID transience) I attempt a connection, the device will then restart advertising after connecting if all of it's connections aren't used. Problem is iOS under the covers seems to claim all available connections to the device when I issue a connect, meaning that a second phone, or keyfob cannot connect. 

This doesn't seem to be an issue on Android and I've tracked it down (can only guess how iOS uses it) to CBCentralManagerScanOptionAllowDuplicatesKey being set to true (in central.swift), with it set to false the problem seems to be gone. I'm not sure of the use case of duplicates when scanning, is there a reason it's set to true in this case? It seems it's false by default and carries a battery life warning - https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerscanoptionallowduplicateskey

Cheers

Anthony

**Comment by farr64** [comment]
The Discussion says that the default is `false`:

```
... If true, the central disables filtering and generates a discovery event each time it receives an advertising packet from the peripheral. 

If false (the default), the central coalesces multiple discoveries of the same peripheral into a single discovery event.
```

Fair enough. However, then Apple issues the warning that @ajkahl mentions:

```
Important
Disabling this filtering can have an adverse effect on battery life; use it only if necessary.
```

So, why would Apple choose a default that has an adverse effect on battery life?

Apple's warning seems to imply the wrong impression that the filtering is `enabled` by Apple by default (as the "wise" choice") and that you, as a developer, should exercise caution when disabling it.

Weird, because Apple clearly says that Apple `disables` this filtering by default. So, the conclusion is that you, as a careful developer, should `enable` this filter.

Strange language choices.

**Reply by farr64** [reply]
Sorry: I meant this as a comment, not as an answer.

**Reply by ajkahl** [reply]
the polarity of the flag makes it confusing, but I read the description as being correct:

- the (apple) default is false, which enables the filter, preventing duplicates, potentially using less power
- it is hard coded to true in the scanning function of this lib, thus disabling the filter and allowing duplicates

**Reply by farr64** [reply]
You are right: Re-re-reading the double negatives clarifies the (obscure) meaning of the whole thing.

As if the entire Bluetooth ecosystem were not convoluted enough.

Many thanks to the authors (and maintainers) of `flutter_reactive_ble` for their brave efforts in making our lives easier.
